test/dependency_source/test_requirement.py

```diff
@@ -94,3 +98,186 @@ def test_requirement_source_duplicate_dependencies(monkeypatch):
     # If the dependency list has duplicates, then converting to a set will reduce the length of the
     # collection
     assert len(specs) == len(set(specs))
+
+
+def _check_fixes(
+    input_reqs: List[str],
+    expected_reqs: List[str],
+    req_paths: List[Path],
+    fixes: List[ResolvedFixVersion],
+) -> None:
+    # Populate the requirements files
+    for (input_req, req_path) in zip(input_reqs, req_paths):
+        with open(req_path, "w") as f:
+            f.write(input_req)
+
+    source = requirement.RequirementSource(req_paths, ResolveLibResolver())
+    for fix in fixes:
+        source.fix(fix)
+
+    # Check the requirements files
+    for (expected_req, req_path) in zip(expected_reqs, req_paths):
+        with open(req_path, "r") as f:
+            assert expected_req == f.read()
+
+
+def test_requirement_source_fix(req_file):
+    _check_fixes(
+        ["flask==0.5\n"],
+        ["flask==1.0\n"],
+        [req_file],
+        [
+            ResolvedFixVersion(
+                dep=ResolvedDependency(name="flask", version=Version("0.5")), version=Version("1.0")
+            )
+        ],
+    )
+
+
+def test_requirement_source_fix_multiple_files(req_file, other_req_file):
+    _check_fixes(
+        ["flask==0.5\n", "requests==1.0\nflask==0.5\n"],
+        ["flask==1.0\n", "requests==1.0\nflask==1.0\n"],
+        [req_file, other_req_file],
+        [
+            ResolvedFixVersion(
+                dep=ResolvedDependency(name="flask", version=Version("0.5")), version=Version("1.0")
+            )
+        ],
+    )
+
+
+def test_requirement_source_fix_specifier_match(req_file, other_req_file):
+    _check_fixes(
+        ["flask<1.0\n", "requests==1.0\nflask<=0.6\n"],
+        ["flask==1.0\n", "requests==1.0\nflask==1.0\n"],
+        [req_file, other_req_file],
+        [
+            ResolvedFixVersion(
+                dep=ResolvedDependency(name="flask", version=Version("0.5")), version=Version("1.0")
+            )
+        ],
+    )
+
+
+def test_requirement_source_fix_specifier_no_match(req_file, other_req_file):
+    # In order to make a fix, the specifier must match the current version and NOT the resolved fix
+    # version. If the specifier matches both, we don't apply the fix since installing from the given
+    # requirements file would already install the fixed version.
+    _check_fixes(
+        ["flask>=0.5\n", "requests==1.0\nflask<2.0\n"],
+        ["flask>=0.5\n", "requests==1.0\nflask<2.0\n"],
+        [req_file, other_req_file],
+        [
+            ResolvedFixVersion(
+                dep=ResolvedDependency(name="flask", version=Version("0.5")), version=Version("1.0")
+            )
+        ],
+    )
+
+
+def test_requirement_source_fix_marker(req_file, other_req_file):
```

Comment:
This test and the one directly below document a few shortcomings of the current approach. Any requirement with markers that don't apply to the current environment are lost. Also, comments are lost as we discussed before.

Reply:
Thanks for adding negative tests for these. We should also make sure to add corresponding tracking issues on the `pip-api` side of things.

Reply:
https://github.com/di/pip-api/issues/120